### PR TITLE
feat(sessions): session tracking & release health (gh-115)

### PR DESCRIPTION
**Pull request by AbianS**
## Summary

- **Ingest**: `session`/`sessions` envelope items dispatched to an in-process `SessionAggregator` (zero per-session DB writes — required for SQLite)
- **Aggregator**: in-memory `HashMap<BucketKey, Counters>` + day-bucketed user dedup; Tokio interval flush + graceful shutdown flush; cardinality cap folding excess releases into `<overflow>`
- **DB**: two new tables — `session_counts` (minute-bucketed) and `session_users` (day-bucketed distinct users); migrations for both Postgres and SQLite
- **API**: `GET /api/projects/{id}/sessions/stats?period=` returning per-release crash-free session/user rates
- **Client**: `SessionsResource.stats()` with Zod schema, exported types
- **MCP**: `get_release_health` tool
- **UI**: `ReleaseHealthCard` server component with color-coded crash-free % rendered on project page

## Test plan

- [ ] `cd apps/server && cargo test` — 109 lib + 138 integration tests pass
- [ ] `cd apps/server && cargo test --no-default-features --features postgres` — Postgres path compiles and passes
- [ ] `pnpm --filter @rustrak/client test` — 327 pass (4 new sessions tests)
- [ ] `pnpm --filter @rustrak/mcp test` — 49 pass (3 new session tool tests + tool-count)
- [ ] `pnpm --filter webview-ui exec tsc --noEmit` — clean
- [ ] Manual: send sessions via `pnpm test-sentry --dsn <dsn> --all`, then `GET .../sessions/stats?period=24h` returns non-zero counts and project page shows the release health card

## Summary by CodeRabbit

* **New Features**
  * Session tracking and per-release "Release Health" metrics
  * New REST API: GET /api/projects/{project_id}/sessions/stats
  * Project dashboard: Release Health card with crash-free rates and session counts
  * Client SDK and MCP/CLI tooling to fetch release health stats

* **Documentation**
  * Spec and OpenAPI docs for session tracking and release health

* **Migrations**
  * DB migrations to add session rollup tables

* **Tests**
  * New unit and integration tests for ingestion, aggregation, API, client, and tools

* **Known Issues**
  * Four minor edge cases: flush risk on partial DB failure, cardinality-cap performance cost, silent/default period parsing, small day-boundary user-window mismatch

### CODE REVIEW

File: apps/server/src/workers/session_aggregator.rs [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+            if let Err(e) = self.upsert_count(key, c).await {
+                log::error!("session_aggregator: flush_counts error: {:?}", e);
+            }
+        }
+        for (key, &crashed) in &users {
+            if let Err(e) = self.upsert_user(key, crashed).await {
+                log::error!("session_aggregator: flush_users error: {:?}", e);
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (AbianS) [reply apps/server/src/workers/session_aggregator.rs:230]: Reviewed: intentional best-effort trade-off for session metrics — same pattern used by Sentry. `std::mem::take` before writes avoids holding the mutex during I/O. `log::error!` provides a visible signal on failure. No code change needed.

File: packages/test-sentry/demo/src/sessions.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    did: 'user-alice',
+    status: 'ok',
+    errors: 0,
+  });
+  await sleep(500);
+  Sentry.captureMessage('Page loaded: /dashboard');
+  await sleep(300);
+  Sentry.captureMessage('Page loaded: /settings');
+  await sleep(300);
+  Sentry.endSession();
```
[the remark this answers is not in the document]
Comment (AbianS) [reply packages/test-sentry/demo/src/sessions.ts:52]: > `Sentry.startSession` / `endSession` don't accept sid/did/status/errors...\n\nReviewed: The SDK type signature at `@sentry/core:exports.d.ts:148` confirms `startSession(context?: SessionContext)` where `SessionContext = Partial<Session>` includes sid, did, status, and errors. These are valid parameters per the official type definitions. No changes needed.
